src/devices/tuya.ts

```diff
@@ -5029,6 +5027,169 @@ const definitions: Definition[] = [
+        onEvent: tuya.onEventSetTime,
+        endpoint: (device) => {
+            return {'l1': 1, 'l2': 1, 'l3': 1, 'state': 1, 'backlight': 1};
+        },
+        meta: {
+            multiEndpoint: true,
+            tuyaDatapoints: [
+                [1, 'state_l1', tuya.valueConverter.onOff],
+                [2, 'state_l2', tuya.valueConverter.onOff],
+                [3, 'state_l3', tuya.valueConverter.onOff],
+                [7, 'countdown_l1', tuya.valueConverter.countdown],
+                [8, 'countdown_l2', tuya.valueConverter.countdown],
+                [9, 'countdown_l3', tuya.valueConverter.countdown],
+                [13, 'state', tuya.valueConverter.onOff],
+                [14, 'power_on_behavior', tuya.valueConverterBasic.lookup({'off': tuya.enum(0), 'on': tuya.enum(1), 'memory': tuya.enum(2)})],
+                [16, 'backlight_mode', tuya.valueConverter.onOff],
+            ],
+        },
+    },
+    {
+        fingerprint: [{modelID: 'TS0601', manufacturerName: '_TZE200_k6jhsr0q'}],
+        model: 'ZS-TYG3-SM-41Z',
+        vendor: 'TuYa',
+        description: 'TuYa 4 Gang Smart Switch with backlight and neutral wire',
+        fromZigbee: [tuya.fz.datapoints],
+        toZigbee: [tuya.tz.datapoints, tuya.tz.power_on_behavior_1],
+        configure: tuya.configureMagicPacket,
+        exposes: [
+            tuya.exposes.switch().withDescription('All Switches'),
+            tuya.exposes.switch().withEndpoint('l1'),
+            tuya.exposes.switch().withEndpoint('l2'),
+            tuya.exposes.switch().withEndpoint('l3'),
+            tuya.exposes.switch().withEndpoint('l4'),
+            tuya.exposes.backlightModeOffOn(),
+            tuya.exposes.countdown().withEndpoint('l1'),
+            tuya.exposes.countdown().withEndpoint('l2'),
+            tuya.exposes.countdown().withEndpoint('l3'),
+            tuya.exposes.countdown().withEndpoint('l4'),
+            e.power_on_behavior(['off', 'on', 'memory']).withAccess(ea.STATE_SET),
+        ],
+        onEvent: tuya.onEventSetTime,
+        endpoint: (device) => {
+            return {'l1': 1, 'l2': 1, 'l3': 1, 'l4': 1, 'state': 1, 'backlight': 1};
+        },
+        meta: {
+            multiEndpoint: true,
+            tuyaDatapoints: [
+                [1, 'state_l1', tuya.valueConverter.onOff],
+                [2, 'state_l2', tuya.valueConverter.onOff],
+                [3, 'state_l3', tuya.valueConverter.onOff],
+                [4, 'state_l4', tuya.valueConverter.onOff],
+                [7, 'countdown_l1', tuya.valueConverter.countdown],
+                [8, 'countdown_l2', tuya.valueConverter.countdown],
+                [9, 'countdown_l3', tuya.valueConverter.countdown],
+                [10, 'countdown_l4', tuya.valueConverter.countdown],
+                [13, 'state', tuya.valueConverter.onOff],
+                [14, 'power_on_behavior', tuya.valueConverterBasic.lookup({'off': tuya.enum(0), 'on': tuya.enum(1), 'memory': tuya.enum(2)})],
```

Comment:
One more thing;

Can you try: `[14, 'power_on_behavior', tuya.valueConverter.powerOnBehavior],` here and instead of `e.power_on_behavior(['off', 'on', 'memory']).withAccess(ea.STATE_SET),` use `e.power_on_behavior()` and see if that works?

Reply:
I can confirm it works.

There is one issue with my implementation, the refresh for backlight_mode throws out the following error:
`error 2023-07-06 18:00:32: No converter available for 'get' 'backlight_mode' ()`
<img width="252" alt="image" src="https://github.com/Koenkk/zigbee-herdsman-converters/assets/5152235/5d92e01d-6461-486e-96a0-c1aee218b649">

Reply:
> Get is not supported for datapoints, use `tuya.exposes.backlightModeOffOn().setAccess(ea.STATE_SET)`

Following error is thrown when I try to use `tuya.exposes.backlightModeOffOn().setAccess(ea.STATE_SET)`

```
node:assert:399
    throw err;
    ^

AssertionError [ERR_ASSERTION]: Does not have any features
    at Binary.setAccess (E:\zigbee2mqtt\zigbee2mqtt\node_modules\zigbee-herdsman-converters\src\lib\exposes.ts:62:15)
    at E:\zigbee2mqtt\zigbee2mqtt\data\extension\externally-loaded.js:27:43
    at Script.runInContext (node:vm:141:12)
    at Script.runInNewContext (node:vm:146:17)
    at Object.runInNewContext (node:vm:300:38)
    at loadModuleFromText (E:\zigbee2mqtt\zigbee2mqtt\lib\util\utils.ts:151:8)
    at loadModuleFromFile (E:\zigbee2mqtt\zigbee2mqtt\lib\util\utils.ts:158:12)
    at Object.getExternalConvertersDefinitions (E:\zigbee2mqtt\zigbee2mqtt\lib\util\utils.ts:168:25)
    at getExternalConvertersDefinitions.next (<anonymous>)
    at new ExternalConverters (E:\zigbee2mqtt\zigbee2mqtt\lib\extension\externalConverters.ts:12:20)
```
I belive this is caused by Binary class constructor does not have features initialized, however I don't know how to use my local copy of zigbee-herdsman-converters to test out this theory.

Reply:
Whoops, it should be `tuya.exposes.backlightModeOffOn().withAccess(ea.STATE_SET)`

Reply:
> I can confirm it works.
> 
> There is one issue with my implementation, the refresh for backlight_mode throws out the following error: `error 2023-07-06 18:00:32: No converter available for 'get' 'backlight_mode' ()` <img alt="image" width="252" src="https://user-images.githubusercontent.com/5152235/251416946-5d92e01d-6461-486e-96a0-c1aee218b649.png">

Upon futher testing, `e.power_on_behavior() ` and `[14, 'power_on_behavior', tuya.valueConverter.powerOnBehavior]` does not work

Reply:
> Whoops, it should be `tuya.exposes.backlightModeOffOn().withAccess(ea.STATE_SET)`

This works